Updated BigBird Model card as per #36979. (#37959)
huggingface/transformers: docs/source/en/model_doc/big_bird.md

">
    </div>
</div>

# BigBird

> [BigBird](https://huggingface.co/papers/2007.14062) is a transformer model built to handle longer sequences — way longer than what models like BERT can manage. Traditional transformers struggle with long inputs because attention gets really expensive as the sequence length grows. BigBird fixes this by using a sparse attention mechanism, which means it doesn’t try to look at everything at once — instead, it mixes in local attention, random attention, and a few global tokens to smartly cover the whole input. This combination gives it the best of both worlds: it keeps the computation efficient while still capturing enough of the sequence to understand it well. Because of this, BigBird is great at tasks involving long documents, like question answering and summarization.

Reviewer: ```suggestion
[BigBird](https://huggingface.co/papers/2007.14062) is a transformer model built to handle sequence lengths up to 4096 compared to 512 for [BERT](./bert). Traditional transformers struggle with long inputs because attention gets really expensive as the sequence length grows. BigBird fixes this by using a sparse attention mechanism, which means it doesn’t try to look at everything at once. Instead, it mixes in local attention, random attention, and a few global tokens to process the whole input. This combination gives it the best of both worlds. It keeps the computation efficient while still capturing enough of the sequence to understand it well. Because of this, BigBird is great at tasks involving long documents, like question answering and summarization.
```

You can find all the original BigBird checkpoints under the [Google](https://huggingface.co/google?search_models=bigbird) organization.

> [!TIP]
> Click on the BigBird models in the right sidebar for more examples of how to apply BigBird to different language tasks.

The example below demonstrates how to predict the `[MASK]` token with [`Pipeline`], [`AutoModel`], and from the command line.

<hfoptions id="usage">
<hfoption id="Pipeline">

```py 
import torch
from transformers import pipeline

pipeline = pipeline(
    task="fill-mask",
    model="google/bigbird-roberta-base",
    torch_dtype=torch.float16,
    device=0
)
pipeline("Plants create [MASK] through a process known as photosynthesis.")
```
</hfoption>
<hfoption id="AutoModel">

```py
import torch
from transformers import AutoModelForMaskedLM, AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained(
    "google/bigbird-roberta-base",
)
model = AutoModelForMaskedLM.from_pretrained(
    "google/bigbird-roberta-base",
    torch_dtype=torch.float16,
    device_map="auto",

>     attention_type ="block_sparse"

Reviewer: No benefit to using it here

```suggestion
```

)
inputs = tokenizer("Plants create [MASK] through a process known as photosynthesis.", return_tensors="pt").to("cuda")

with torch.no_grad():
    outputs = model(**inputs)
    predictions = outputs.logits

masked_index = torch.where(inputs['input_ids'] == tokenizer.mask_token_id)[1]
predicted_token_id = predictions[0, masked_index].argmax(dim=-1)
predicted_token = tokenizer.decode(predicted_token_id)

print(f"The predicted token is: {predicted_token}")
```

</hfoption>
<hfoption id="transformers CLI">

```bash

> !echo -e "Plants create [MASK] through a process known as photosynthesis." | transformers-cli run --task fill-mask --model google/bigbird-roberta-base --device 0

Reviewer: ```suggestion
!echo -e "Plants create [MASK] through a process known as photosynthesis." | transformers run --task fill-mask --model google/bigbird-roberta-base --device 0
```

Author: > 

Hi @stevhliu, in a previous PR (for BART) you mentioned that I should leave the transformers-cli cmd as is, should I do that here as well? I might have mis understood what you meant.

Reviewer: Yeah, sorry about the misunderstanding. This should still be `transformers-cli run ...`. It is only simplified for the `chat` command where you can do `transformers chat ...`

```
</hfoption>
</hfoptions>

## Notes

> - Inputs should be padded on the right, since BigBird uses absolute position embeddings.

Reviewer: ```suggestion
- Inputs should be padded on the right because BigBird uses absolute position embeddings.
```

> - If the input sequence is too short, the block_sparse attention type will automatically fall back to original_full attention, since sparse patterns don’t offer much benefit for small inputs.

Reviewer: ```suggestion
- BigBird supports `original_full` and `block_sparse` attention. If the input sequence length is less than 1024, it is recommended to use `original_full` since sparse patterns don't offer much benefit for smaller inputs.
- The current implementation uses window size of 3 blocks and 2 global blocks, only supports the ITC-implementation, and doesn't support `num_random_blocks=0`.
- The sequence length must be divisible by the block size.

## Resources

- Read the [BigBird](https://huggingface.co/blog/big-bird) blog post for more details about how its attention works.
```
